nodl_python/nodl/types/_types.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2020 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the
+# GNU Limited General Public License version 3, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranties of MERCHANTABILITY, SATISFACTORY QUALITY, or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Limited General Public License for more details.
+#
+# You should have received a copy of the GNU Limited General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+from abc import ABC
+from typing import Any, Dict, List
+
+from nodl._util import qos_to_dict
+from rclpy.qos import QoSPresetProfiles, QoSProfile
+
+
+class NoDLData(ABC):
+    """Data structure base class for NoDL objects."""
+
+    def __repr__(self) -> str:
+        return str(self._as_dict)
+
+    def __str__(self) -> str:
+        return str(self._as_dict)
+
+    @property
+    def _as_dict(self) -> Dict:
```

Comment:
I might be missing it, but this functionality doesn't appear to be used anywhere in this PR. What is its purpose?

Reply:
This should be part of the cli PR, this chunk got staged incorrectly.

Reply:
addressed in 97698c7

Reply:
This PR was made to add features that would be used the the cli, displaying nodes in human-readable form is one the verbs.